### ISSUE TITLE
Add ability to modify the `ServiceLifetime` on AddSqlServerDbContext

### ISSUE THREAD

**Issue by benjaminsampica** [dotnet/aspire#1194]
In some contexts, the service lifetime of the DbContext registration needs [to not be the default of `Scoped`](https://learn.microsoft.com/en-us/ef/core/dbcontext-configuration/#implicitly-sharing-dbcontext-instances-via-dependency-injection), which is what the default `AddDbContext<>` registration lifetime is. Please let me modify the lifetime easily.

**Comment by eerhardt** [comment]
Seems like this could be a new property on https://github.com/dotnet/aspire/blob/07722afc66a940e10e0ba49ecd282e0fdf3584bc/src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/NpgsqlEntityFrameworkCorePostgreSQLSettings.cs#L9

(and others) 
just like the `bool DbContextPooling` we have today.

**Comment by eerhardt** [comment]
This should be fixed when #1403 is done.

**Comment by sebastienros** [comment]
@benjaminsampica this has been merged, here is a sample

```c#
        builder.Services.AddDbContext<ITestDbContext, TestDbContext>(optionsBuilder =>
        {
            optionsBuilder.UseSqlServer(ConnectionString);
        }, contextLifetime: ServiceLifetime.Singleton);

        builder.EnrichSqlServerDbContext<TestDbContext>();
```
